src/ocamlorg_web/lib/handler.ml

```diff
@@ -637,7 +637,26 @@ let packages_search t req =
     Dream.from_percent_encoded
       (match Dream.query req "q" with Some search -> search | None -> "")
   in
-  let results = List.map (Package_helper.frontend_package t) current_items in
+
+  let open Lwt.Infix in
+  let documentation_status (pkg : Ocamlorg_package.t) =
+    Ocamlorg_package.documentation_status ~kind:`Package pkg
+    >|= fun package_documentation_status ->
+    match package_documentation_status with
+    | Some { failed = false; _ } -> Ocamlorg_frontend.Package.Success
+    | Some { failed = true; _ } -> Failure
+    | None -> Unknown
+  in
+
+  let open Lwt.Syntax in
+  let* results =
+    Lwt_list.map_p
+      (fun pkg ->
+        documentation_status pkg >|= fun doc_status ->
```

Comment:
```suggestion
        documentation_status pkg >|= fun doc_status ->
```

Maybe this can be written without the `>|=` operator. We try to keep things very basic, preferring let* and let+ over these operators.

Reply:
got it, I'll try to do it without the Infix.